src/MSBuildBinLogQuery/Graph/ProjectNode.cs

```diff
@@ -13,9 +16,9 @@ public class ProjectNode : IQueryableGraphNode
         public ItemManager Items { get; }
         public PropertyManager Properties { get; }
         public PropertyManager GlobalProperties { get; }
-        public List<TargetNode> ExecutedTargets { get; }
-        public List<TargetNode> EntryPointTargets { get; }
-        public HashSet<ProjectNode> ProjectsBeforeThis { get; }
+        public ConcurrentDictionary<string, TargetNode> Targets { get; }
+        public ImmutableHashSet<TargetNode> EntryPointTargets { get; }
```

Comment:
I think this should be an `ImmutableList`, even though managing it will be harder. The order can be relevant (in some cases).

_Really_ it should be an unordered set of ordered lists, but let's hope we never have to be that accurate.

Reply:
Done. Actually I made it to an `IReadOnlyList`.